iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergOutputCommitter.java

```diff
@@ -459,9 +474,14 @@ private void commitTable(FileIO io, ExecutorService executor, OutputTable output
       deleteFiles.addAll(writeResults.deleteFiles());
       replacedDataFiles.addAll(writeResults.replacedDataFiles());
       referencedDataFiles.addAll(writeResults.referencedDataFiles());
+      mergedAndDeletedFiles.addAll(writeResults.mergedAndDeletedFiles());
     }
 
-    FilesForCommit filesForCommit = new FilesForCommit(dataFiles, deleteFiles, replacedDataFiles, referencedDataFiles);
+    dataFiles.removeIf(dataFile -> mergedAndDeletedFiles.contains(new Path(String.valueOf(dataFile.path()))));
```

Comment:
why don't we filter when adding stuff? why is that cross table thing?

Reply:
While writing data, there are multiple `jobContexts`. The files from one jobContext can act as the input files for merge task which are written in another jobContexts. Hence to resolve them, its done during commit phase.

Reply:
what I meant is
````
mergedAndDeletedFiles.addAll(writeResults.mergedAndDeletedFiles());
dataFiles.remove(writeResults.mergedAndDeletedFiles())
deleteFiles.remove(writeResults.mergedAndDeletedFiles())
````

Reply:
Not so easy since we maintain the list of DataFile objects and set of paths.